mesonbuild/compilers/compilers.py

```diff
@@ -438,37 +440,41 @@ class CompilerArgs(list):
     # In generate_link() we add external libs without de-dup, but we must
     # *always* de-dup these because they're special arguments to the linker
     always_dedup_args = tuple('-l' + lib for lib in unixy_compiler_internal_libs)
-    compiler = None
-
-    def _check_args(self, args):
-        cargs = []
-        if len(args) > 2:
-            raise TypeError("CompilerArgs() only accepts at most 2 arguments: "
-                            "The compiler, and optionally an initial list")
-        elif not args:
-            return cargs
-        elif len(args) == 1:
-            if isinstance(args[0], (Compiler, StaticLinker)):
-                self.compiler = args[0]
-            else:
-                raise TypeError("you must pass a Compiler instance as one of "
-                                "the arguments")
-        elif len(args) == 2:
-            if isinstance(args[0], (Compiler, StaticLinker)):
-                self.compiler = args[0]
-                cargs = args[1]
-            elif isinstance(args[1], (Compiler, StaticLinker)):
-                cargs = args[0]
-                self.compiler = args[1]
-            else:
-                raise TypeError("you must pass a Compiler instance as one of "
-                                "the two arguments")
-        else:
-            raise AssertionError('Not reached')
-        return cargs
 
-    def __init__(self, *args):
-        super().__init__(self._check_args(args))
+    def __init__(self, compiler: typing.Union['Compiler', StaticLinker],
+                 iterable: typing.Optional[typing.Iterable[str]] = None):
+        self.compiler = compiler
+        self.__container = list(iterable) if iterable is not None else []  # type: typing.List[str]
+
+    @typing.overload
+    def __getitem__(self, index: int) -> str: ...
```

Comment:
__[Flake8]__ multiple statements on one line (def) ([see on Sider](https://sider.review/gh/repos/19784232/pulls/6295#issue-4904881))

<details>

| Rule |
| - |
| E704 |

</details>
<sub>posted by Sider</sub>

Reply:
This is is the upstream recommended way to do this, but I'll go ahead and use pass instead so that flake8 will shut up.